src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddleware.cs

```diff
@@ -116,7 +118,7 @@ private async Task HandleException(HttpContext context, ExceptionDispatchInfo ed
                 context.Response.StatusCode = StatusCodes.Status500InternalServerError;
                 context.Response.OnStarting(_clearCacheHeadersDelegate, context.Response);
 
-                await _options.ExceptionHandler(context);
+                await _options.ExceptionHandler!(context);
```

Comment:
💡 Consider adding a property (even if private) for the class which returns `_options.ExceptionHandler` as a known non-null value

Reply:
I tried doing this, but it felt like it made the options type more complex for minimal benefit. I appreciate the feedback but I'll leave this one be.